### PR TITLE
update code for ARM module and new distro

### PR DESCRIPTION
**Pull request by Nidylei**
update code for ARM module and new distro

### CODE REVIEW

File: TestLibs/RDFELibs.psm1 [hunk 1]
```diff
@@ -79,7 +79,11 @@ Function DetectLinuxDistro($VIP, $SSHport, $testVMUser, $testVMPassword)
 	}
 	else
 	{
-		if ($DistroName -imatch "UBUNTU")
+		if ($DistroName -imatch "Ubuntu15.10")
```
Comment (lizzha) [inline TestLibs/RDFELibs.psm1]: It might break other cases, possibly there's a script that reference the value

File: TestLibs/ARMLibrary.psm1 [hunk 2]
```diff
@@ -196,6 +196,8 @@ Function GenerateAzureDeployJSONFile ($RGName, $osImage, $osVHD, $RGXMLData, $Lo
 {
 $jsonFile = $azuredeployJSONFilePath
 $StorageAccountName = $xml.config.Azure.General.ARMStorageAccount
+$StorageAccountType = $xml.config.Azure.General.AccountType
```
Comment (iamshital) [inline TestLibs/ARMLibrary.psm1]: I don't think that, we need to mention the storage account type to be hardcoded in XML file. We can get the storage type from PS commands.

File: TestLibs/RDFELibs.psm1 (hunk 1 above)
Comment (lizzha) [inline TestLibs/RDFELibs.psm1]: you could leave here unchanged, and use the value of the global "detectedDistro"

File: SetupScripts/DetectLinuxDistro.sh [hunk 3]
```diff
@@ -16,7 +16,8 @@ while echo $1 | grep ^- > /dev/null; do
     shift
 done
         if [ -e /etc/debian_version ]; then
-                echo "UBUNTU"
+                #echo "UBUNTU"
+				echo `python -c 'import platform;print platform.dist()[0]+ platform.dist()[1]'`
```
Comment (lizzha) [inline SetupScripts/DetectLinuxDistro.sh]: suggest to comply with the convention in this script, check the file content of "/etc/os-release"

File: TestLibs/RDFELibs.psm1 [hunk 4]
```diff
@@ -79,7 +79,7 @@ Function DetectLinuxDistro($VIP, $SSHport, $testVMUser, $testVMPassword)
 	}
 	else
 	{
-		if ($DistroName -imatch "UBUNTU")
+		elseif ($DistroName -imatch "UBUNTU")
```
Comment (lizzha) [inline TestLibs/RDFELibs.psm1]: if

File: remote-scripts/SETUP-INSTALL-PACKAGES.ps1 [hunk 5]
```diff
@@ -64,15 +64,28 @@ if ($isDeployed)
...
+					#VM De-provision
                     if ($DistroName -eq "COREOS")   
                     {  
                         $output = RunLinuxCmd -username $user -password $password -ip $hs1VIP -port $hs1vm1sshport -command "/usr/share/oem/python/bin/python /usr/share/oem/bin/waagent -force -deprovision+user 2>&1" -runAsSudo  
-                    }  
+                    }
                     else {  
                         GetVMLogs -DeployedServices $isDeployed 
-                        $output = RunLinuxCmd -username $user -password $password -ip $hs1VIP -port $hs1vm1sshport -command "/usr/sbin/waagent -force -deprovision+user 2>&1" -runAsSudo  
+						if($usePython3)
```
Comment (lizzha) [inline remote-scripts/SETUP-INSTALL-PACKAGES.ps1]: Please pay attention to indent

File: TestLibs/ARMLibrary.psm1 (hunk 2 above)
Comment (Nidylei) [inline TestLibs/ARMLibrary.psm1]: thanks for your suggestion,I will update it later.

File: remote-scripts/azuremodules.py [hunk 6]
```diff
@@ -45,7 +45,7 @@ def UpdateRepos(current_distro):
 	if ((current_distro == "ubuntu") or (current_distro == "Debian")):
 		Run("apt-get update")
 	elif ((current_distro == "rhel") or (current_distro == "Oracle") or (current_distro == 'centos')):
-		Run("yum -y update")
+		RunYumUpdate("yum -y update")
```
Comment (lizzha) [inline remote-scripts/azuremodules.py]: the issue may also exists when you do "apt-get update" or "zypper update", when the image is quite old and there're lots of update packages to install. Suggest to name the method as "RunUpdate", and also call this method if it's ubuntu or suse

File: remote-scripts/azuremodules.py [hunk 7]
```diff
@@ -169,6 +169,28 @@ def Run(cmd):
...
+            #RunLog.error(op)
+            exception = 1
+            #updateState('TestFailed')
+        else:
+            #RunLog.info(op)
+            #print (op)
+            return op
+        if exception == 1:
+            str_code = str(code)
+            #RunLog.critical("Exception, return code is " + str_code + #" for command " + cmd)
```
Comment (lizzha) [inline remote-scripts/azuremodules.py]: please remove the unnecessary comment code

File: remote-scripts/azuremodules.py (hunk 6 above)
Comment (lizzha) [inline remote-scripts/azuremodules.py]: please add a comment to describe the issue we're trying to fix so that people could know why we make the change

**Pull request merged**